SU2_CFD/src/driver_structure.cpp

```diff
@@ -362,16 +409,26 @@ CDriver::CDriver(char* confFile,
 
   for (iZone = 0; iZone < nZone; iZone++) {
 
+    grid_movement[iZone] = new CVolumetricMovement*[nInst[iZone]];
+    for (iInst = 0; iInst < nInst[iZone]; iInst++)
+      grid_movement[iZone][iInst] = NULL;
+
     if (config_container[iZone]->GetGrid_Movement() ||
         (config_container[iZone]->GetDirectDiff() == D_DESIGN)) {
       if (rank == MASTER_NODE)
-        cout << "Setting dynamic mesh structure for zone "<< iZone<<"." << endl;
-      grid_movement[iZone] = new CVolumetricMovement(geometry_container[iZone][MESH_0], config_container[iZone]);
+        cout << "Setting dynamic mesh structure for zone "<< iZone + 1<<"." << endl;
+      for (iInst = 0; iInst < nInst[iZone]; iInst++){
+        grid_movement[iZone][iInst] = new CVolumetricMovement(geometry_container[iZone][iInst][MESH_0], config_container[iZone]);
+      }
       FFDBox[iZone] = new CFreeFormDefBox*[MAX_NUMBER_FFD];
       surface_movement[iZone] = new CSurfaceMovement();
-      surface_movement[iZone]->CopyBoundary(geometry_container[iZone][MESH_0], config_container[iZone]);
-      if (config_container[iZone]->GetUnsteady_Simulation() == HARMONIC_BALANCE)
-        iteration_container[iZone]->SetGrid_Movement(geometry_container, surface_movement, grid_movement, FFDBox, solver_container, config_container, iZone, 0, 0);
+      surface_movement[iZone]->CopyBoundary(geometry_container[iZone][INST_0][MESH_0], config_container[iZone]);
+      if (config_container[iZone]->GetUnsteady_Simulation() == HARMONIC_BALANCE){
```

Comment:
Is there a way to remove these "if" statements?

Reply:
Our goal should be reducing if statements to a minimum. This if statement was already there, however, and in this PR, I tried to minimize impact on anyone's work.